Actions/ReadSettings/ReadSettings.ps1

```diff
@@ -49,48 +49,47 @@ try {
     }
 
     if ($settings.versioningstrategy -ne -1) {
-        if ($getSettings -contains 'appBuild' -or $getSettings -contains 'appRevision') {
-            switch ($settings.versioningStrategy -band 15) {
-                0 { # Use RUN_NUMBER and RUN_ATTEMPT
-                    $settings.appBuild = $settings.runNumberOffset + [Int32]($ENV:GITHUB_RUN_NUMBER)
-                    $settings.appRevision = [Int32]($ENV:GITHUB_RUN_ATTEMPT) - 1
-                }
-                1 { # Use RUN_ID and RUN_ATTEMPT
-                    OutputError -message "Versioning strategy 1 is no longer supported"
-                }
-                2 { # USE DATETIME
-                    $settings.appBuild = [Int32]([DateTime]::UtcNow.ToString('yyyyMMdd'))
-                    $settings.appRevision = [Int32]([DateTime]::UtcNow.ToString('HHmmss'))
-                }
-                15 { # Use maxValue
-                    $settings.appBuild = [Int32]::MaxValue
-                    $settings.appRevision = 0
-                }
-                default {
-                    OutputError -message "Unknown version strategy $versionStrategy"
-                    exit
-                }
+        switch ($settings.versioningStrategy -band 15) {
+            0 { # Use RUN_NUMBER and RUN_ATTEMPT
+                $settings.appBuild = $settings.runNumberOffset + [Int32]($ENV:GITHUB_RUN_NUMBER)
+                $settings.appRevision = [Int32]($ENV:GITHUB_RUN_ATTEMPT) - 1
+            }
+            1 { # Use RUN_ID and RUN_ATTEMPT
+                OutputError -message "Versioning strategy 1 is no longer supported"
+            }
+            2 { # USE DATETIME
+                $settings.appBuild = [Int32]([DateTime]::UtcNow.ToString('yyyyMMdd'))
```

Comment:
[Out of context]

If a pipeline (say, CICD) runs around midnight, then artifacts will have different version. Is this by design?

Reply:
If you have project with dependencies or like and you decide to use strategy 2 - then yes.